### PR TITLE
[13.0][ADD] helpdesk_mgmt_sla 13.0.1.1.0

### PR DESCRIPTION
**Pull request by hgarridoco**
This PR aims to add SLA.

Last try to PR from this repo was wrong because is my first PR to OCA and I failed.

### CODE REVIEW

**Review by marcelsavegnago — COMMENTED** [review]
Great job. However, here are some comments.

- The SLA record could allow you to be associated with multiple teams.
- In the team registration, it would include a tab to select the applied SLAs or even create a new one.
- Including demo data is also a good thing.
- I didn't see data about the SLA on the ticket form.
- After everything is evaluated and reviewed, I would appreciate it if you could do a PR for the V12 as well.

I noticed that you are calculating / changing the SLA expiration date on closed and canceled tickets.
image

**Comment by hgarridoco** [comment]
@marcelsavegnago I'll make your comments, you're right!

**Comment by hgarridoco** [comment]
@marcelsavegnago check it out!

**Comment by marcelsavegnago** [comment]
@hgarridoco it'd be good to have a few unit tests :)

**Review by marcelsavegnago — COMMENTED** [review]
@hgarridoco it'd be good to have a few unit tests :)

**Comment by DarioLodeiros** [comment]
@hgarridoco can you review the create_datetime field on create method? Travis fails becouse this field does not exist

File: helpdesk_mgmt_sla/tests/test_helpdesk_mgmt_sla.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                "hours": 2,
+            }
+        )
+
+    def generate_ticket(self):
+        return self.env["helpdesk.ticket"].create(
+            {
+                "name": "Test Ticket 1",
+                "description": "Test ticket description",
+                "team_id": [(6, 0, [self.team_id.id])],
```
Comment (DarioLodeiros) [inline helpdesk_mgmt_sla/tests/test_helpdesk_mgmt_sla.py]: Integer Field, not recordset

**Review by DarioLodeiros — CHANGES_REQUESTED** [review]
The code still has bugs, you can check them in the travis log

File: helpdesk_mgmt_sla/tests/test_helpdesk_mgmt_sla.py (hunk 1 above)
Comment (hgarridoco) [reply helpdesk_mgmt_sla/tests/test_helpdesk_mgmt_sla.py]: Changed!

**Comment by hgarridoco** [comment]
@DarioLodeiros please check this new commit and review it.
@marcelsavegnago sorry about my delay, lot of work!

**Review by marcelsavegnago — APPROVED** [review]
LGTM

**Review by marcelsavegnago — APPROVED** [review]
LGTM

**Review by Saran440 — COMMENTED** [review]
Functional Test.
- SLA Deadline is generate mistake. If I config SLA is 2 hours, it generate SLA Deadline + 1 day and 16.00 only.

- I config SLA Reach Stage is `in progress`, should it start SLA when stage ticket is `in progress` ? but I create new ticket stage `new` it start SLA deadline or I mistake something?
Selection_006


**Results**
Selection_007

**Comment by hgarridoco** [comment]
> Functional Test.
> 
> * SLA Deadline is generate mistake. If I config SLA is 2 hours, it generate SLA Deadline + 1 day and 16.00 only.
> * I config SLA Reach Stage is `in progress`, should it start SLA when stage ticket is `in progress` ? but I create new ticket stage `new` it start SLA deadline or I mistake something?
>   Selection_006
> 
> **Results**
> Selection_007

Hi,

Be careful to check if deadline is correct because the code check resource calender to calculate it. For example, if some ticket arrive at 12 pm and your resource calendar is from 8 am to 17 pm, and your deadline horus are 2, deadline will be set at 10 am.

About reach stage is which stage you need to move this ticket to not calculate deadline. For example, if I need that my tickets arrives at  "In progress" in 2 hours or less, I will set this stage at reach stage.

This module calculate SLA like Odoo EE.

Sorry about if I take some tome to answer your question or problems, I'm so bussy.

**Comment by hgarridoco** [comment]
@DarioLodeiros can you review it again?

**Review by spgevers — APPROVED** [review]
Functional test successful.

**Comment by marcelsavegnago** [comment]
ping @OCA/helpdesk-maintainers

**Comment by marcelsavegnago** [comment]
@hgarridoco can you squash the commits?

**Review by Saran440 — APPROVED** [review]
Okay I see. Functional Test :+1:

**Comment by hgarridoco** [comment]
> @hgarridoco can you squash the commits?

Done!

**Comment by marcelsavegnago** [comment]
@hgarridoco  I didn't find anything in the code that evaluates the reach stage configured in the SLA.

**Comment by marcelsavegnago** [comment]
Not that it's for this PR, but I have some demands to include in the SLA configuration:
* Minimum priority 
* Categories 
* Tags

**Comment by elvise** [comment]
@DarioLodeiros any chance for this PR ?

**Pull request merged**